x/poolmanager/router_test.go

```diff
@@ -1296,6 +1299,170 @@ func (suite *KeeperTestSuite) TestSingleSwapExactAmountIn() {
 	}
 }
 
+type MockPoolModule struct {
+	pools []types.PoolI
+}
+
+func (m *MockPoolModule) GetPools(ctx sdk.Context) ([]types.PoolI, error) {
+	return m.pools, nil
+}
+
+// This test suite contains test cases for the AllPools function, which returns a sorted list of all pools from different pool modules.
```

Comment:
I am good with all the logic in this PR, but as Matt said I am not quite sure why we aren't creating all three pool types for this test case (and not utilizing the mocks)

Reply:
Replied in Matt's thread about this. Going to resolve this one